src/atopile/components.py

```diff
@@ -81,66 +71,112 @@ class NoMatchingComponent(errors.AtoError):
     title = "No component matches parameters"
 
 
-@cache
-def _get_generic_from_db(component_addr: str) -> dict:
+# Define the cache file path
+repo = Repo(".", search_parent_directories=True)
+top_level_path = Path(repo.working_tree_dir)
+cache_file_path = top_level_path / ".ato/component_cache.json"
+
+# Try to load the cache, if it exists
+if cache_file_path.exists():
+    with open(cache_file_path, "r") as cache_file:
+        component_cache = json.load(cache_file)
+else:
+    component_cache = {}
+
+
+def save_cache():
+    """Saves the current state of the cache to a file."""
+    with open(cache_file_path, "w") as cache_file:
+        # Convert the ChainMap to a regular dictionary
+        serializable_cache = dict(component_cache)
+        json.dump(serializable_cache, cache_file)
+
+
+def has_component_changed(cached_entry, current_data):
+    """Check if the component data has changed based on the address."""
+    # Implement logic to compare relevant parts of current_data with cached_entry['address_data']
+    # Return True if data has changed, False otherwise
+    if current_data != cached_entry["address_data"]:
+        log.info("Component data has changed for updating cache")
+        return True
+    return False
+
+
+def get_component_from_cache(component_addr, current_data):
+    """Retrieve a component from the cache, if available, not stale, and unchanged."""
+    cached_entry = component_cache.get(component_addr)
+    if cached_entry:
+        cached_timestamp = datetime.fromtimestamp(cached_entry["timestamp"])
+        if datetime.now() - cached_timestamp < timedelta(
+            days=1
+        ) and not has_component_changed(cached_entry, current_data):
+            return cached_entry["data"]
+    return None
+
+
+def update_cache(component_addr, component_data, address_data):
+    """Update the cache with new component data and save it."""
+    component_cache[component_addr] = {
+        "data": component_data,
+        "timestamp": time.time(),  # Current time as a timestamp
+        "address_data": dict(address_data),  # Data used to detect changes
+    }
+    save_cache()
+
+
+def clean_cache():
+    """Clean out entries older than 1 day."""
+    for addr, entry in list(component_cache.items()):
+        cached_timestamp = datetime.fromtimestamp(entry["timestamp"])
+        if datetime.now() - cached_timestamp >= timedelta(days=1):
+            del component_cache[addr]
+    save_cache()
+
+import logging
```

Comment:
These should probably be at the top of the file - they're currently casuing linting failures